src/openrct2/ride/Ride.cpp

```diff
@@ -5484,22 +5484,36 @@ void ride_get_start_of_track(CoordsXYE * output)
 {
     track_begin_end trackBeginEnd;
     CoordsXYE trackElement = *output;
-    if (track_block_get_previous(trackElement.x, trackElement.y, trackElement.element, &trackBeginEnd)) {
+    if (track_block_get_previous(trackElement.x, trackElement.y, trackElement.element, &trackBeginEnd))
+    {
         rct_tile_element* initial_map = trackElement.element;
+        track_begin_end slowIt = trackBeginEnd;
+        bool moveSlowIt = false;
         do {
             CoordsXYE lastGood = {
                 /* .x = */ trackBeginEnd.begin_x,
                 /* .y = */ trackBeginEnd.begin_y,
                 /* .element = */ trackBeginEnd.begin_element
             };
 
-            if (!track_block_get_previous(trackBeginEnd.end_x, trackBeginEnd.end_y, trackBeginEnd.begin_element, &trackBeginEnd)) {
+            if (!track_block_get_previous(trackBeginEnd.end_x, trackBeginEnd.end_y, trackBeginEnd.begin_element, &trackBeginEnd))
+            {
                 trackElement = lastGood;
                 break;
             }
+
+            moveSlowIt = !moveSlowIt;
```

Comment:
does moveSlowIt mean it's slowly interacting through the track pieces?

Reply:
_iterating_. Yes, to detect loops you need two iterators, `fast` and `slow` one, you iterate them until `fast` reaches end of the list or the two meet.

Reply:
that must have been my phones spell check.